Connections to Neo4j

mnmami opened the pull request:
It's identical to JDO, I got the code from there, and tested it. In JDO, it should work for MariaDB also, but since I didn't test there I just wrote "ENABLE FOR MySQL", but I believe it will work also.

mnmami committed:
Add connection to MySQL

It's identical to JDO, I got the code from there, and tested it. In JDO, it should work for MariaDB also, but since I didn't test there I just wrote "ENABLE FOR MySQL", but I believe it will work also.

renataogarcia commented on tutorial/src/main/resources/META-INF/persistence.xml:
```diff
@@ -23,6 +23,13 @@
             <property name="javax.persistence.jdbc.user" value="sa"/>
             <property name="javax.persistence.jdbc.password" value=""/-->
 
+            <!-- ENABLE THESE FOR MySQL -->
```
Should be ok to have MariaDB too.

renataogarcia commented on tutorial/src/main/resources/META-INF/persistence.xml:
```diff
@@ -23,6 +23,13 @@
+            <property name="javax.persistence.jdbc.driver" value="com.mysql.jdbc.Driver" />   
+            <property name="javax.persistence.jdbc.user" value="root" />   
+            <property name="javax.persistence.jdbc.password" value="password" />  
+            <property name="javax.jpa.option.Mapping" value="mysql" />-->
```
This is a JDO feature, so we don't need this line(31).

renataogarcia commented:
Thanks! Just did two minor comments before I can merge it.

mnmami committed:
Value of Neo4j connector

The actual value "neo4j:testDB" isn't clear for beginners. I read some people asking what to put there (as I asked myself). I suggest the semantic "/path/to/" which is common in such situations. I suggest to put the default Neo4j graph database folder "Graph.db", I believe these changes will save new developers time and are more appropriate for Neo4j case.
Thanks to do the same thing for other stores if it has the same logic, namely, a database directory and not an SGBD like MySQL. To be general, we may also apply this notation for one-file databases like *.xls, so it becomes "/path/to/tutorial.xls".

renataogarcia commented:
Sorry, but can we use different issues/pull requests for the MySQL and Neo4j changes? The MySQL changes can be merged once the comments I've made above are addressed.

Regarding the Neo4j change, it is a good suggestion, but we also need to change the build which is cleaning these files each run. See [here](https://github.com/MohamedNadjibMAMI/samples-jpa/blob/master/tutorial/pom.xml#L197) for instance.

andyjefferson commented:
I merged most of this in to GitHub master ... but instead of changing lines to things that won't work out of the box I put a comment "change this to whatever your path to your graph database is" etc. It should always work out of the box wherever possible
